localstack/services/iam/provider.py

```diff
@@ -297,6 +308,80 @@ def get_service_linked_role_deletion_status(
         # TODO: test
         return GetServiceLinkedRoleDeletionStatusResponse(Status=DeletionTaskStatusType.SUCCEEDED)
 
+    def put_user_permissions_boundary(
+        self, context: RequestContext, user_name: userNameType, permissions_boundary: arnType
+    ) -> None:
+        if user := moto_iam_backend.users.get(user_name):
+            user.permissions_boundary = permissions_boundary
+        else:
+            raise NoSuchEntityException()
+
+    def delete_user_permissions_boundary(
+        self, context: RequestContext, user_name: userNameType
+    ) -> None:
+        if user := moto_iam_backend.users.get(user_name):
+            if hasattr(user, "permissions_boundary"):
+                delattr(user, "permissions_boundary")
+        else:
+            raise NoSuchEntityException()
+
+    def create_user(
+        self,
+        context: RequestContext,
+        user_name: userNameType,
+        path: pathType = None,
+        permissions_boundary: arnType = None,
+        tags: tagListType = None,
+    ) -> CreateUserResponse:
+        response = call_moto(context=context)
+        user = moto_iam_backend.get_user(user_name)
+        if permissions_boundary:
+            user.permissions_boundary = permissions_boundary
+            response["User"]["PermissionsBoundary"] = AttachedPermissionsBoundary(
+                PermissionsBoundaryArn=permissions_boundary,
+                PermissionsBoundaryType="Policy",
+            )
+        return response
+
+    def get_user(
+        self, context: RequestContext, user_name: existingUserNameType = None
+    ) -> GetUserResponse:
+        response = call_moto(context=context)
+        moto_user_name = response["User"]["UserName"]
+        moto_user = moto_iam_backend.users.get(moto_user_name)
+        # if the user does not exist or is no user
+        if not moto_user:
+            access_key_id = extract_access_key_id_from_auth_header(context.request.headers)
+            sts_client = aws_stack.connect_to_service(
+                "sts",
+                aws_access_key_id=access_key_id,
+                aws_secret_access_key=INTERNAL_AWS_SECRET_ACCESS_KEY,
+            )
+            caller_identity = sts_client.get_caller_identity()
```

Comment:
curious about this part. Is this behavior something you managed to observe in AWS as well?

Is the `INTERNAL_AWS_SECRET_ACCESS_KEY` intended to be used here to differentiate between inter-service requests that should *not* be checked & tracked? :thinking: 
Previously I think we always assumed that all internal calls had `__internal_call__` for both access key & secret

Reply:
Here, I abuse the sts service get_caller_identity to get the caller principal (arn), to distinguish the case where we are root (so not a "user" per se, but still valid for this call) and the case where we call as an assumed-role or something similar, where the operation just won't work.
For this, I only set the access key id of the caller (since we do not have the secret key anyways), and set the secret key to internal (could be anything really). This should be reworked anyway as part of the inter service communication.